src/blog/2025-12-01-type-aware-alpha.md

```diff
@@ -0,0 +1,249 @@
+
+## Quick Start
+
+Install `oxlint` and `oxlint-tsgolint`, then run with the `--type-aware` flag:
+
+::: code-group
+
+```sh [npm]
+npm add -D oxlint oxlint-tsgolint@latest
+npx oxlint --type-aware
+```
+
+```sh [pnpm]
+pnpm add -D oxlint oxlint-tsgolint@latest
+pnpm oxlint --type-aware
+```
+
+```sh [yarn]
+yarn add -D oxlint oxlint-tsgolint@latest
+yarn oxlint --type-aware
+```
+
+```sh [bun]
+bun add -D oxlint oxlint-tsgolint@latest
+bunx oxlint --type-aware
+```
+
+:::
+
+To try a specific type-aware rule without other configuration (`oxlint-tsgolint` must be installed globally or locally already):
+
+::: code-group
+
+```sh [npm]
+npx oxlint --type-aware -A all -D typescript/no-floating-promises
+```
+
+```sh [pnpm]
+pnpx oxlint --type-aware -A all -D typescript/no-floating-promises
+```
+
+```sh [yarn]
+yarn oxlint --type-aware -A all -D typescript/no-floating-promises
+```
+
+```sh [bun]
+bunx oxlint --type-aware -A all -D typescript/no-floating-promises
+```
+
+:::
+
+For more configuration options, see our [usage guide](/docs/guide/usage/linter/type-aware).
+
+## Performance
+
+| Project         | Oxlint + Type Aware | ESLint + typescript-eslint | Improvement |
+| --------------- | ------------------- | -------------------------- | ----------- |
+| vuejs/core      | 2.531 s             | 20.800 s                   | 8.22x       |
+| outline/outline | 4.448 s             | 55.070 s                   | 12.38x      |
+
+Benchmarks were performed on a MacBook Pro M2 Max 12 Cores (8 performance and 4 efficiency).
+
+Our performance testing shows that `oxlint` with type-aware linting is around 10 times faster than `eslint` with `typescript-eslint`. Take a look at our [performance benchmarks](https://github.com/oxc-project/bench-linter) for more details.
+
+Oxlint can also be used to typecheck your codebase while linting. This avoids a duplicate work, as much of the type information is already computed during type-aware linting..
+
+:::warning Known Issues
+While `tsgolint` is ready for testing in production codebases, you may encounter issues with running out of memory when working with very large codebases. We are working on optimizing the memory usage for the next milestone. We would love if you tried `tsgolint` and reported any out-of-memory issues to us in the [`tsgolint` repository](https://github.com/oxc-project/tsgolint) and included some details about your project to help us improve memory usage.
+:::
+
+## What's new since the technical preview?
+
+### Support for type-checking while linting
+
+`tsgolint` now supports emitting type checking errors from TypeScript while linting. Since type-aware rules already require checking all of the types within a file, we are able to use this existing type information rather than discarding it. This means that in some cases, it is possible to skip doing a separate type-check command altogether (e.g., `tsc --noEmit`), reducing total time spent doing linting and type-checking in CI.
```

Comment:
Should maybe note that the typechecking behavior may differ from TS5/6 in some edge cases and link to the tsgo section later in the post? I know when I've tested tsgo on my work repo we've had a handful of errors that show up because the location where the `@ts-expect-error` is necessary has changed, due to the diagnostics not being perfectly 1:1 with TypeScript 5.